scrapy/core/spidermw.py

```diff
@@ -61,7 +61,7 @@ def _process_spider_input(self, scrape_func: ScrapeFunc, response: Response, req
             try:
                 result = method(response=response, spider=spider)
                 if result is not None:
-                    msg = (f"Middleware {method.__qualname__} must return None "
```

Comment:
Since it is a middleware method, not a middleware, “Method” or “Middleware method” would be the right choice instead of “Middleware”. However, I don’t think we need a word at all, we are already providing the qualified path.